st3/sublime_lib/resource_path.py

```diff
@@ -18,7 +21,148 @@ def glob_resources(pattern):
+    that the global root is the empty string, not `/`. All resource paths are
+    absolute; dots in paths have no special meaning.
+
+    Like :class:`pathlib.Path` objects, :class:`ResourcePath` objects are
+    immutable, hashable, and orderable with each other. The forward slash
+    operator is a shorthand for :meth:`joinpath`. The string representation
+    of a :class:`ResourcePath` is the raw resource path in the form that Sublime
+    Text uses."""
+
+    def __init__(self, *pathsegments):
+        self._parts = tuple(
+            part
+            for segment in pathsegments if segment
+            for part in posixpath.normpath(str(segment)).split('/')
+        )
+
+    def __hash__(self):
+        return hash(self.parts)
+
+    def __repr__(self):
+        return "{}({!r})".format(self.__class__.__name__, str(self))
+
+    def __str__(self):
+        return '/'.join(self.parts)
+
+    def __eq__(self, other):
+        return isinstance(other, ResourcePath) and self._parts == other.parts
+
+    def __lt__(self, other):
+        if isinstance(other, ResourcePath):
+            return self._parts < other.parts
+        else:
+            return NotImplemented
+
+    def __truediv__(self, other):
+        return self.joinpath(other)
+
+    @property
+    def parts(self):
+        """A tuple giving access to the path’s various components."""
+        return self._parts
+
+    @property
+    def parent(self):
+        """The logical parent of the path. The empty path is its own parent."""
+        if len(self._parts) > 1:
+            return self.__class__(*self._parts[:-1])
+        else:
+            return self
+
+    @property
+    def parents(self):
+        """An immutable sequence providing access to the logical ancestors of
+        the path."""
+        parent = self.parent
+        if self == parent:
+            return ()
+        else:
+            return (parent,) + parent.parents
+
+    @property
+    def name(self):
+        """A string representing the final path component, or the empty string
+        if the path is empty."""
+        try:
+            return self._parts[-1]
+        except IndexError:
+            return ''
+
+    @property
+    def suffix(self):
+        """The file extension of the final component, or the empty string if the
+        path is empty or if the final component does not have a file extension.
+        """
+        return posixpath.splitext(self.name)[1]
+
+    @property
+    def suffixes(self):
+        """A list of the path’s file extensions."""
+        return ['.' + suffix for suffix in self.name.split('.')[1:]]
+
+    @property
+    def stem(self):
+        """The final path component without its suffix, or the empty string if
+        the path is empty."""
+        return posixpath.splitext(self.name)[0]
+
+    @property
+    def root(self):
+        """The first path component (usually `Packages` or `Cache`)."""
```

Comment:
The values should be in quotes since they are strings (within code blocks).

Also, sphinx uses reST, which requires two backticks for inline code.

Reply:
Done.